interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/Annotations.kt

```diff
@@ -39,15 +43,34 @@ public fun KSAnnotation.toAnnotationSpec(): AnnotationSpec {
   // TODO support type params once they're exposed https://github.com/google/ksp/issues/753
   for (argument in arguments) {
     val value = argument.value ?: continue
-    val member = CodeBlock.builder()
     val name = argument.name!!.getShortName()
+    if (omitDefaultValues) {
+      val defaultValue = this.defaultArguments.firstOrNull { it.name?.asString() == name }?.value
+      if (isDefaultValue(value, defaultValue)) { continue }
+    }
+    val member = CodeBlock.builder()
     member.add("%N = ", name)
-    addValueToBlock(value, member)
+    addValueToBlock(value, member, omitDefaultValues)
     builder.addMember(member.build())
   }
   return builder.build()
 }
 
+private fun isDefaultValue(value: Any?, defaultValue: Any?): Boolean {
+  if (defaultValue == null) return false
+  if (value is KSAnnotation && defaultValue is KSAnnotation) {
+    return defaultValue.defaultArguments.all { defaultValueArg ->
+      isDefaultValue(value.arguments.firstOrNull { it.name == defaultValueArg.name }?.value, defaultValueArg.value)
+    }
+  }
+  if (value is List<*> && defaultValue is List<*>) {
+    return value.size == defaultValue.size && (0 until defaultValue.size).all { index ->
+      isDefaultValue(value[index], defaultValue[index])
+    }
+  }
+  return Objects.deepEquals(value, defaultValue)
```

Comment:
I wonder if there's a different API we can use, if we ever want to get this code to run without the JDK, having `java.*` dependencies will make it harder.

Reply:
replaced by simple `value == defaultValue`